driver/src/deposit_driver.rs

```diff
@@ -17,29 +19,30 @@ where
         find_first_unapplied_slot(deposit_slot, &|i| contract.has_deposit_slot_been_applied(i))?;
     if slot <= deposit_slot {
         info!("Highest unprocessed deposit_slot is {:?}", slot);
-        let creation_time_block = |i| contract.creation_timestamp_for_deposit_slot(i);
-        if can_process(slot, contract, &creation_time_block)? {
-            info!("Processing deposit_slot {:?}", slot);
-            let state_root = contract.get_current_state_root()?;
-            let contract_deposit_hash = contract.deposit_hash_for_slot(slot)?;
-            let mut balances = db.get_balances_for_state_root(&state_root)?;
-
-            let deposits = db.get_deposits_of_slot(&slot)?;
-            let deposit_hash = deposits.rolling_hash(0);
-            hash_consistency_check(deposit_hash, contract_deposit_hash, "deposit")?;
-
-            balances.apply_deposits(&deposits);
-
-            info!("New AccountState hash is {}", balances.state_hash);
-            contract.apply_deposits(
-                slot,
-                state_root,
-                balances.state_hash,
-                contract_deposit_hash,
-            )?;
-            return Ok(true);
-        } else {
-            info!("Need to wait before processing deposit_slot {:?}", slot);
+        match batch_processing_state(slot, contract, &|i| {
```

Comment:
nit: Maybe is simpler to read if we do one thing in each line.  Like:

```rust
let processing_state = batch_processing_state(slot, contract, &|i| {
    contract.creation_timestamp_for_deposit_slot(i)
})?;

match processing_state {
   // ...
}
```

Reply:
sweet. I like it